sys/net/application_layer/dhcpv6/client.c

```diff
@@ -716,6 +848,100 @@ static bool _parse_reply(uint8_t *rep, size_t len)
                     }
                 }
                 break;
+#if IS_USED(MODULE_GNRC_DHCPV6_CLIENT_IA_NA)
+            case DHCPV6_OPT_IA_NA:
+                for (unsigned i = 0; i < CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX; i++) {
```

Comment:
Same here as above
```suggestion
            case DHCPV6_OPT_IA_NA:
                for (unsigned i = 0;
                    IS_USED(MODULE_GNRC_DHCPV6_CLIENT_IA_NA) &&
                    (i < CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX);
                    i++) {
```

Reply:
Should be addressed by 98d3cd1fe47ba4a1800e28a5890852172815c9ca